Modules/SubModule_Import/xASL_bids_BIDSifyCheckTimeEncoded.m

```diff
@@ -0,0 +1,90 @@
+%   bTimeEncoded    - Boolean describing if the current sequence is a time encoded sequence
+%   bTimeEncodedFME - Boolean describing if the current sequence is a specific FME time encoded sequence
+%
+% -----------------------------------------------------------------------------------------------------------------------------------------------------
+% DESCRIPTION: Check for time encoded sequence.
+%
+% EXAMPLE:     [jsonIn,jsonOut,bTimeEncoded,bTimeEncodedFME] = xASL_bids_BIDSifyCheckTimeEncoded(jsonIn,jsonOut);
+%
+% __________________________________
+% Copyright 2015-2021 ExploreASL
+
+    % Check for time encoded sequences
+    if isfield(jsonOut,'TimeEncodedMatrixSize') && ~isempty(jsonOut.TimeEncodedMatrixSize) || ... % Should be 4, 8 or 12
+        isfield(jsonOut,'TimeEncodedMatrixType') % Natural or walsh
+        bTimeEncoded = true; 
+    else 
+        bTimeEncoded = false; 
+    end
+
+
+    % Check for specific time encoded sequence of FME (Fraunhofer Mevis)
+    if isfield(jsonIn,'SeriesDescription')
+        bTimeEncodedFME = ~isempty(regexp(jsonIn.SeriesDescription,'(Encoded_Images_Had)\d\d(_)\d\d(_TIs_)\d\d(_TEs)', 'once'));
+    else
+        bTimeEncodedFME = false;
+    end
+
+    if bTimeEncodedFME
+        bTimeEncoded = true;    
+    end
+
+    if bTimeEncoded
+        if isfield(jsonOut,'EchoTime') && isfield(jsonOut,'PostLabelingDelay')
+            % From the import, the length of EchoTime should correspond to the number of volumes
+            if length(jsonOut.EchoTime)~=length(jsonOut.PostLabelingDelay)
+                % So here, we first make sure that each PLD is repeated for the whole block of echo-times
+                jsonOut.TimeEncodedNumberTE = length(uniquetol(jsonOut.EchoTime,0.001)); % Obtain the number of echo times
+                repeatedPLDs = repmat(jsonOut.PostLabelingDelay(:),1,jsonOut.TimeEncodedNumberTE)';
+                repeatedPLDs = repeatedPLDs(:);
+
+                if length(repeatedPLDs) > length(jsonOut.EchoTime) || mod(length(jsonOut.EchoTime),length(repeatedPLDs)) ~= 0
+                    warning('Did not succeed in repeating PLDs for each TE for Hadamard sequence import...');
+                end
+                % Make sure that number of volumes can be divided by the repeated PLDs
+                jsonOut.PostLabelingDelay = repeatedPLDs;
+            end
+        end
+    end
+
+    % Check for FME Hadamard sequences
+    if isfield(jsonOut,'SeriesDescription')
+        bTimeEncodedFME = ~isempty(regexp(jsonOut.SeriesDescription,'(Encoded_Images_Had)\d\d(_)\d\d(_TIs_)\d\d(_TEs)', 'once'));
+        if bTimeEncodedFME
+            startDetails = regexp(jsonOut.SeriesDescription,'\d\d(_)\d\d(_TIs_)\d\d(_TEs)', 'once');
+            jsonOut.TimeEncodedMatrixSize = xASL_str2num(jsonOut.SeriesDescription(startDetails:startDetails+1));
+            jsonOut.TimeEncodedNumberTI = xASL_str2num(jsonOut.SeriesDescription(startDetails+3:startDetails+4));
```

Comment:
Same as for NumberTE
At line 43,  "numberTIs = 1"
everywhere, "jsonOut.TimeEncodedNumberTI" -> "numberTIs"
line 76, "isfield(jsonOut,'TimeEncodedNumberTI')" -> "numberTIs>1"
line 78 with TEs delete
line 79 "NumberPLDs = jsonOut.TimeEncodedNumberTI + 1;" -> "NumberPLDs = NumberTIs + 1;"

Reply:
a57d26f 👍